### DISCUSSION TITLE
Oracle has released an open source Python 'Thin' driver that connects directly to the DB

### DISCUSSION THREAD

**Discussion by cjbj** [sijms/go-ora#209] [Show and tell]
We (Oracle) just pushed a major release of our Python driver for Oracle Database.  It has a new name: [python-oracledb](https://oracle.github.io/python-oracledb/).  You might be interested in its 'Thin' mode code, available under an open source licence.  See the [repo](https://github.com/oracle/python-oracledb), in particular see https://github.com/oracle/python-oracledb/tree/main/src/oracledb/impl/thin which implements direct connection to Oracle Database.

**Comment by hmmftg** [comment]
our experience tells us "don't trust oracle ever again"

thanks

**Reply by domgiles** [reply]
Sorry to hear that. I'd be keen to hear about your experiences and what we can do to regain that trust. Please feel free to contact me directly.

**Comment by sijms** [comment]
ok no problem I will see the code 
thank you for sharing this

**Comment by sijms** [comment]
do you have implementation for "DES40C" encryption used in encryption service because I don't find it in your source or other modern  driver

**Reply by cjbj** [reply]
Internal discussions about what we can, and can't, put into open source in this area need to occur first.  The current solution for Python users is to enable the Thick mode which uses the Oracle Client libraries.

**Comment by sijms** [comment]
no problem. the issue solved you can review it #210
